### ISSUE TITLE
BUG: run.sh script in router sidecar

### ISSUE THREAD

**Issue by naseemkullah** [oracle/mysql-operator#195]
## Is this a BUG REPORT or FEATURE REQUEST?

BUG REPORT 

## Versions

**MySQL Operator Version**: 0.2.0

**Environment**:
- **Kubernetes version** (use `kubectl version`): 1.10.5
- **Cloud provider or hardware configuration**: GKE on GCP
- **OS** (e.g. from /etc/os-release): COS
- **Kernel** (e.g. `uname -a`): ?
- **Others**: n/a

## What happened?

Deployed wordpress router example as per https://github.com/oracle/mysql-operator/tree/master/examples/demo/wordpress-router

And then checked logs of router sidecar and got:
```
$ kubectl logs wordpress-router-76d64584f9-wrqkz mysqlrouter
+ exec /run.sh mysqlrouter
Succesfully contacted mysql server at mysql-wordpress-0.mysql-wordpress. Checking for cluster state.
/run.sh: line 41: [: : integer expression expected
Waiting for 3 cluster instances to become available via mysql-wordpress-0.mysql-wordpress (0/12)
Waiting for 3 cluster instances to become available via mysql-wordpress-0.mysql-wordpress (1/12)
/run.sh: line 41: [: : integer expression expected
Waiting for 3 cluster instances to become available via mysql-wordpress-0.mysql-wordpress (2/12)
/run.sh: line 41: [: : integer expression expected
/run.sh: line 41: [: : integer expression expected
Waiting for 3 cluster instances to become available via mysql-wordpress-0.mysql-wordpress (3/12)
Waiting for 3 cluster instances to become available via mysql-wordpress-0.mysql-wordpress (4/12)
/run.sh: line 41: [: : integer expression expected
Waiting for 3 cluster instances to become available via mysql-wordpress-0.mysql-wordpress (5/12)
/run.sh: line 41: [: : integer expression expected
/run.sh: line 41: [: : integer expression expected
Waiting for 3 cluster instances to become available via mysql-wordpress-0.mysql-wordpress (6/12)
Waiting for 3 cluster instances to become available via mysql-wordpress-0.mysql-wordpress (7/12)
/run.sh: line 41: [: : integer expression expected
/run.sh: line 41: [: : integer expression expected
Waiting for 3 cluster instances to become available via mysql-wordpress-0.mysql-wordpress (8/12)
/run.sh: line 41: [: : integer expression expected
Waiting for 3 cluster instances to become available via mysql-wordpress-0.mysql-wordpress (9/12)
/run.sh: line 41: [: : integer expression expected
Waiting for 3 cluster instances to become available via mysql-wordpress-0.mysql-wordpress (10/12)
/run.sh: line 41: [: : integer expression expected
Waiting for 3 cluster instances to become available via mysql-wordpress-0.mysql-wordpress (11/12)
```

## What you expected to happen?

No `/run.sh: line 41: [: : integer expression expected`

## How to reproduce it (as minimally and precisely as possible)?

Deploy the wordpress router example and check logs of router sidecar.

## Anything else we need to know?

Please let me know if anything is unclear and thank you!

**Comment by 941zturn** [comment]
I got the same issue yesterday. I found the root cause was I had changed password secret. The "new" password did not changed in mysql database. While mysql router using the new password to connect to the cluster. That's why it can not get the cluster status.

**Comment by neumayer** [comment]
Without being a specialist here, I don't think the kubernetes behaviour is "wrong" here because it seems that changes to environment variables (even via secrets) do not retrigger redeployment.

https://kubernetes.io/docs/concepts/configuration/secret/

What I can do though, is improve the error message on the router container end, i.e. check if a connection can be established first before polling the cluster member numbers and exit with an error message in that case.

Sound reasonable?

**Comment by prydie** [comment]
@naseemkullah I believe the root cause of the issue was the root password Secret reference being incorrectly set in the example Cluster yaml. I've fixed it in #205.

@neumayer It would certainly be nice though to have the router handle incorrect passwords with better user feedback 🙂
